sqlglot/dialects/clickhouse.py

```diff
@@ -84,6 +84,39 @@ class Parser(parser.Parser):
 
         LOG_DEFAULTS_TO_LN = True
 
+        def _parse_placeholder(self) -> t.Optional[exp.Expression]:
+            """
+            Parse a placeholder expression like SELECT {abc: UInt32} or FROM {table: Identifier}
+            https://clickhouse.com/docs/en/sql-reference/syntax#defining-and-using-query-parameters
+            """
+            index = self._index
+
+            if self._match(TokenType.L_BRACE):
+                # We might be getting a variable from a SELECT clause, or potentially an identifier from a FROM clause
+                if not self._match_set({TokenType.VAR, TokenType.TABLE}):
+                    return None
+
+                this = self._prev.text
+                # Ensure we've got a colon delimiting the placeholder name and placeholder type
+                if not self._match(TokenType.COLON):
+                    return None
+                data_type: t.Optional[t.Union[str, exp.Expression]] = self._parse_types()
+                # If we couldn't get the type, we could be looking at the "Identifier" literal
+                if not data_type:
+                    identifier_seq = "Identifier"
+                    if not self._match_text_seq(identifier_seq.upper()):
+                        return None
+                    data_type = identifier_seq
```

Comment:
I think this can be reduce to just `data_type = self._parse_type()`.

Reply:
The reason I didn't go with `_parse_type()` here is because that has `check_funcs=True` which converts the `Map` type to a function instead of the `Map` type

e.g. 
```- SELECT {abc: UInt32}, {b: String}, {c: DateTime}, {d: map(String, [UInt8])}, {e: Tuple(UInt8, String)}
?                                                       ^           ^     ^
+ SELECT {abc: UInt32}, {b: String}, {c: DateTime}, {d: Map(String, Array(UInt8))}, {e: Tuple(UInt8, String)}
?                                                       ^           ^^^^^^     ^
```

Reply:
Perhaps I should just update the dialect around `Map` as a type vs. `map` as a function